rust/blockstore/src/arrow/ordered_blockfile_writer.rs

```diff
@@ -128,9 +129,7 @@ impl ArrowOrderedBlockfileWriter {
     pub(crate) async fn commit<K: ArrowWriteableKey, V: ArrowWriteableValue>(
         mut self,
     ) -> Result<ArrowBlockfileFlusher, Box<dyn ChromaError>> {
-        let mut inner = Arc::into_inner(self.inner)
-            .expect("There should be no other strong references to the completed block deltas")
-            .into_inner();
+        let mut inner = std::mem::take(&mut *self.inner.lock().await);
```

Comment:
Why this change?

Reply:
there may be multiple outstanding strong references to `self.inner` since `ArrowOrderedBlockfileWriter` is cloneable

Reply:
This should never happen though in the way we use it.

Reply:
there is right now in tests, I'll try to track down where it's happening

Reply:
Yeah but not in our real code, ideally we'd modify the tests, IIRC @sanketkedia  did not do mem take for a reason.

Reply:
The unordered writer does not do `Arc::into_inner()` either, it mutates `self` instead of taking ownership.

`MetadataSegmentWriter` is cloned for every partition. `MetadataSegmentWriter` holds a `FullTextIndexWriter` which holds a `BlockfileWriter`. I believe the root cause is that the `FlushS3Operator` takes a `MetadataSegmentWriter` in its `&input`, but `.clone()`s the `MetadataSegmentWriter` in its `.run()` because operators take an immutable reference to input instead of taking ownership of the input.

Reply:
I suppose we could wrap the `BlockfileWriter` in an `Arc` inside `FullTextIndexWriter` but that seems rather hacky. Are you ok leaving as-is for now and addressing this on both the unordered and ordered writers together in the future?